### DISCUSSION TITLE
Is it possible to train with the help of a colab of an Ubuntu virtual machine connected locally to the computer when using Jupiter?

### DISCUSSION THREAD

**Discussion by Maximus-show** [TheLastBen/fast-stable-diffusion#1306] [General]
Hello Ben. Thanks for your work. Can you please tell me if I can work in your colab by connecting to my PC using Jupiter ? I tried and I got a problem. I hope for your help. Thank you :)

**Comment by TheLastBen** [comment]
You can, but you need to change the folder paths in the colab, and install the dependencies from this repo, and make sure you have enough VRAM

**Reply by Maximus-show** [reply]
I have a card with 12GB of video memory, is that enough?
I registered the path, but I could make a mistake somewhere because I don’t see the model.
I seem to have installed the dependencies, but I can’t be sure if I installed everything because there is no requirements.txt file in the github

**Reply by MiJiw** [reply]
Hi Maximus, I managed to run the notebook on my local workstation (Ubuntu 20.04, dual RTX3090) with minor changes to the hardcoded paths.

Here is the complete list of packages that you can copy paste to a requirements.txt : 

absl-py==1.3.0
accelerate==0.12.0
aiohttp==3.8.3
aiosignal==1.3.1
asttokens==2.2.1
async-timeout==4.0.2
attrs==22.2.0
backcall==0.2.0
bitsandbytes==0.36.0.post2
cachetools==5.2.0
capture==0.1
certifi==2022.12.7
charset-normalizer==2.1.1
cmake==3.25.0
comm==0.1.2
datasets==2.8.0
debugpy==1.6.5
decorator==5.1.1
dill==0.3.6
entrypoints==0.4
executing==1.2.0
filelock==3.9.0
frozenlist==1.3.3
fsspec==2022.11.0
git-lfs==1.6
google-auth==2.15.0
google-auth-oauthlib==0.4.6
grpcio==1.51.1
huggingface-hub==0.11.1
idna==3.4
importlib-metadata==6.0.0
ipykernel==6.19.4
ipython==8.8.0
jedi==0.18.2
jupyter_client==7.4.8
jupyter_core==5.1.2
Markdown==3.4.1
MarkupSafe==2.1.1
matplotlib-inline==0.1.6
multidict==6.0.4
multiprocess==0.70.14
mypy-extensions==0.4.3
nest-asyncio==1.5.6
numpy==1.24.1
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
oauthlib==3.2.2
packaging==22.0
pandas==1.5.2
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.4.0
platformdirs==2.6.2
prompt-toolkit==3.0.36
protobuf==3.20.3
psutil==5.9.4
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==10.0.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
Pygments==2.14.0
pyre-extensions==0.0.23
python-dateutil==2.8.2
pytz==2022.7
PyYAML==6.0
pyzmq==24.0.1
regex==2022.10.31
requests==2.28.1
requests-oauthlib==1.3.1
responses==0.18.0
rsa==4.9
six==1.16.0
stack-data==0.6.2
tensorboard==2.11.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tokenizers==0.13.2
torch==1.13.1
torchvision==0.14.1
tornado==6.2
tqdm==4.64.1
traitlets==5.8.0
transformers==4.25.1
triton==2.0.0.dev20221120
typing-inspect==0.8.0
typing_extensions==4.4.0
urllib3==1.26.13
wcwidth==0.2.5
Werkzeug==2.2.2
wget==3.2
xformers==0.0.16rc402
xxhash==3.2.0
yarl==1.8.2
zipp==3.11.0

**Reply by Maximus-show** [reply]
> Hi Maximus, I managed to run the notebook on my local workstation (Ubuntu 20.04, dual RTX3090) with minor changes to the hardcoded paths.
> 
> Here is the complete list of packages that you can copy paste to a requirements.txt :
> 
> absl-py==1.3.0 accelerate==0.12.0 aiohttp==3.8.3 aiosignal==1.3.1 asttokens==2.2.1 async-timeout==4.0.2 attrs==22.2.0 backcall==0.2.0 bitsandbytes==0.36.0.post2 cachetools==5.2.0 capture==0.1 certifi==2022.12.7 charset-normalizer==2.1.1 cmake==3.25.0 comm==0.1.2 datasets==2.8.0 debugpy==1.6.5 decorator==5.1.1 dill==0.3.6 entrypoints==0.4 executing==1.2.0 filelock==3.9.0 frozenlist==1.3.3 fsspec==2022.11.0 git-lfs==1.6 google-auth==2.15.0 google-auth-oauthlib==0.4.6 grpcio==1.51.1 huggingface-hub==0.11.1 idna==3.4 importlib-metadata==6.0.0 ipykernel==6.19.4 ipython==8.8.0 jedi==0.18.2 jupyter_client==7.4.8 jupyter_core==5.1.2 Markdown==3.4.1 MarkupSafe==2.1.1 matplotlib-inline==0.1.6 multidict==6.0.4 multiprocess==0.70.14 mypy-extensions==0.4.3 nest-asyncio==1.5.6 numpy==1.24.1 nvidia-cublas-cu11==11.10.3.66 nvidia-cuda-nvrtc-cu11==11.7.99 nvidia-cuda-runtime-cu11==11.7.99 nvidia-cudnn-cu11==8.5.0.96 oauthlib==3.2.2 packaging==22.0 pandas==1.5.2 parso==0.8.3 pexpect==4.8.0 pickleshare==0.7.5 Pillow==9.4.0 platformdirs==2.6.2 prompt-toolkit==3.0.36 protobuf==3.20.3 psutil==5.9.4 ptyprocess==0.7.0 pure-eval==0.2.2 pyarrow==10.0.1 pyasn1==0.4.8 pyasn1-modules==0.2.8 Pygments==2.14.0 pyre-extensions==0.0.23 python-dateutil==2.8.2 pytz==2022.7 PyYAML==6.0 pyzmq==24.0.1 regex==2022.10.31 requests==2.28.1 requests-oauthlib==1.3.1 responses==0.18.0 rsa==4.9 six==1.16.0 stack-data==0.6.2 tensorboard==2.11.0 tensorboard-data-server==0.6.1 tensorboard-plugin-wit==1.8.1 tokenizers==0.13.2 torch==1.13.1 torchvision==0.14.1 tornado==6.2 tqdm==4.64.1 traitlets==5.8.0 transformers==4.25.1 triton==2.0.0.dev20221120 typing-inspect==0.8.0 typing_extensions==4.4.0 urllib3==1.26.13 wcwidth==0.2.5 Werkzeug==2.2.2 wget==3.2 xformers==0.0.16rc402 xxhash==3.2.0 yarl==1.8.2 zipp==3.11.0

Thanks, dear friend. Can you please tell me which version you are using. python and CUDA and CuDnn ?

**Reply by Maximus-show** [reply]
> Вы можете, но вам нужно изменить пути к папкам в colab, установить зависимости из этого репозитория и убедиться, что у вас достаточно видеопамяти

![image](https://user-images.githubusercontent.com/121389128/211251374-051ae7d8-22e1-4ccb-acbe-7d89f5c81f8d.png)
No model found, use the "Model Download" cell to download a model

**Reply by MiJiw** [reply]
> Thanks, dear friend. Can you please tell me which version you are using. python and CUDA and CuDnn ?
Sure, I use : 
Python 3.9.12
CUDA 11.7
cuDNN 8.5

I might have a bad news though, I checked nvidia-smi and you must have at least 18Gb of VRAM in order to run it locally (with modifications to support xformers in the custom train_dreambooth.py script of TheLastBen).

**Reply by Maximus-show** [reply]
> > Спасибо, дорогой друг. Не могли бы вы сказать мне, какую версию вы используете. python, CUDA и cuDNN?Конечно, я использую: Python 3.9.12 CUDA 11.7 cuDNN 8.5
> 
> Хотя у меня могут быть плохие новости, я проверил nvidia-smi, и у вас должно быть не менее 18 ГБ видеопамяти, чтобы запустить его локально (с изменениями для поддержки xformers в пользовательском train_dreambooth.py сценарий последнего дня).

thank you. I was hoping I could get by with 12 GB. looks like I'll have to postpone this question until I can afford a better card. thanks for the help
